src/dask_awkward/__init__.py

```diff
@@ -16,7 +16,11 @@
     partition_compatibility,
 )
 from dask_awkward.lib.describe import fields
-from dask_awkward.lib.inspect import necessary_columns, sample
```

Comment:
I think it makes sense to (for now) keep a `dak.necessary_columns` that just calls `report_necessary_columns`; we should deprecate it, but this has been in for a little while now and it might be best to keep it around for a while. We've never had a formal deprecation process in dask-awkward, but now might be a time to start!